<a href="https://colab.research.google.com/github/deniwidi/data-science-2026/blob/main/Pertemuan2_Deni_Widi_Alfian_240401010340.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [26]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

In [27]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


**Jawaban Praktikum Pertemuan 2**

**Q1. Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?**

In [28]:
print(f"Shape  : {df.shape}")
print(f"Baris  : {df.shape[0]}")
print(f"Kolom  : {df.shape[1]}")
print("\nDaftar kolom:")
for i, col in enumerate(df.columns, 1):
    print(f"  {i:2}. {col}")

Shape  : (891, 12)
Baris  : 891
Kolom  : 12

Daftar kolom:
   1. PassengerId
   2. Survived
   3. Pclass
   4. Name
   5. Sex
   6. Age
   7. SibSp
   8. Parch
   9. Ticket
  10. Fare
  11. Cabin
  12. Embarked


Jawaban:

Dataset Titanic memiliki 891 baris dan 12 kolom.

---



**Q2. Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?**

In [29]:
missing = df.isnull().sum()
percent = (missing / len(df)) * 100

pd.DataFrame({
    'Missing': missing,
    'Percent (%)': percent
})

,Missing,Percent (%)
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


Jawaban:

Berikut adalah kolom yang memiliki missing value:

1. Kolom Cabin, sebanyak 687 baris kosong (77.1%)
2. Kolom Age,sebanyak 177 baris kosong (19.9%)
3. Kolom Embarked, sebanyak 2 baris kosong (0.2%)


---



**Q3. Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy
atau Pandas.**

In [30]:
# Menggunakan NumPy
survived_arr = df['Survived'].to_numpy()
pct_selamat = np.mean(survived_arr) * 100
print(f"Selamat    : {np.sum(survived_arr)} orang")
print(f"Tidak      : {np.sum(survived_arr == 0)} orang")
print(f"% Selamat  : {pct_selamat:.2f}%")

# Verifikasi dengan Pandas
print(f"Pandas check: {df['Survived'].mean()*100:.2f}%")

Selamat    : 342 orang
Tidak      : 549 orang
% Selamat  : 38.38%
Pandas check: 38.38%


Jawaban:

Dari dataset tersebut, hanya 38.38% dari 891 penumpang yang selamat.

---




**Q4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?**

In [31]:
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

jumlah = len(wanita_kelas1)
rata_usia = wanita_kelas1['Age'].mean()
min_usia = wanita_kelas1['Age'].min()
max_usia = wanita_kelas1['Age'].max()

print(f"Jumlah penumpang wanita kelas 1        : {jumlah} penumpang")
print(f"Rata-rata usia penumpang wanita kelas 1: {rata_usia:.1f} tahun")

Jumlah penumpang wanita kelas 1        : 94 penumpang
Rata-rata usia penumpang wanita kelas 1: 34.6 tahun


Jawaban:

Dari dataset tersebut, penumpang wanita dari kelas 1 sebanyak 94 penumpang, dengan rata-rata usia 34.6 tahun (~35 tahun).


---



**Q5. Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?**

In [32]:
survival_by_class = df.groupby('Pclass')['Survived'].mean()
survival_by_class

,Survived
Pclass,
1,0.629630
2,0.472826
3,0.242363


Jawaban:

Dari dataset tersebut, kelas 1 memiliki tingkat keselamatan tertinggi (62.96%), hampir 2.5× lebih tinggi dari Kelas 3 (24.24%).

---



**Q6. Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.**

In [33]:
# Analisis multi-variabel: Sex × Pclass
pivot = df.groupby(['Pclass', 'Sex'])['Survived']\
          .mean().unstack().round(3) * 100
print(pivot)

Sex     female  male
Pclass              
1         96.8  36.9
2         92.1  15.7
3         50.0  13.5


Jawaban:

Dari dataset tersebut, wanita di semua kelas jauh lebih selamat daripada pria. Pria kelas 2 (15.7%) bahkan lebih rendah dari wanita kelas 3 (50%).